internal/common/function/function.go

```diff
@@ -280,6 +281,41 @@ func CheckVeleroBackupLabels(backup *velerov1api.Backup) bool {
 	return exists && value == constant.ManagedByLabelValue
 }
 
+// DeleteVeleroBackup deletes Velero Backup object based on the NonAdminBackup object name and namespace
+func DeleteVeleroBackup(ctx context.Context, r client.Client, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	veleroBackupName := GenerateVeleroBackupName(nab.Namespace, nab.Name)
+
+	logger.V(1).Info(fmt.Sprintf("Attempting to delete VeleroBackup: %s", veleroBackupName))
+	veleroBackup := velerov1api.Backup{}
+
+	err := r.Get(ctx, types.NamespacedName{Name: veleroBackupName, Namespace: constant.OadpNamespace}, &veleroBackup)
+
+	if err == nil {
+		err = r.Delete(ctx, &veleroBackup)
+		if err != nil {
+			logger.V(1).Info(fmt.Sprintf("Error deleting VeleroBackup: %s", veleroBackupName))
+			return false, err
+		}
+	} else if apierrors.IsNotFound(err) {
+		return false, nil
+	}
```

Comment:
I think implementation is flawed here

we ignore case that `err` is not nil and different than `apierrors.IsNotFound`, not?

that means that Velero Backup was not even retrieved, but we say it got deleted

Reply:
a suggestion is to always check if `err != nil` when dealing with errors to avoid forgetting anything

Reply:
100% agree on that, will add that case.